code/modules/cooking/cooking_container.dm

```diff
@@ -0,0 +1,307 @@
+/obj/effect/cooking_container_lip
+	icon = 'icons/obj/cooking/containers.dmi'
+	icon_state = "blank"
+	vis_flags = VIS_INHERIT_ID
+
+/**
+ * Cooking containers are used in ovens, fryers and so on, to hold multiple
+ * ingredients for a recipe. They interact with the cooking process, and link
+ * up with the cooking code dynamically. Originally sourced from the Aurora,
+ * heavily retooled to actually work with PCWJ Holder for a portion of an
+ * incomplete meal, allows a cook to temporarily offload recipes to work on
+ * things factory-style, eliminating the need for 20 plates to get things done
+ * fast.
+ **/
+/obj/item/reagent_containers/cooking
+	icon = 'icons/obj/cooking/containers.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	volume = 240
+	var/datum/cooking/recipe_tracker/tracker = null
+	/// Icon state of the lip layer of the object
+	var/lip
+	var/obj/effect/cooking_container_lip/lip_effect
+	/// A flat quality reduction for removing an unfinished recipe from the container.
+	var/removal_penalty = 0
+	container_type = OPENCONTAINER
+	/// Record of what cooking has been done on this food.
+	var/list/cooker_data = list()
+	var/claimed = FALSE
+	var/preposition = "In"
+	visible_transfer_rate = null
+	possible_transfer_amounts = null
+
+	new_attack_chain = TRUE
+
+/obj/item/reagent_containers/cooking/Initialize(mapload)
+	. = ..()
+	flags |= REAGENT_NOREACT
+	if(lip)
+		lip_effect = new
+		lip_effect.icon_state = lip
+
+	clear_cooking_data()
+
+/obj/item/reagent_containers/cooking/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>You can <b>Alt-Click</b> to remove items from this.</span>"
+
+	if(length(contents))
+		. += get_content_info()
+
+/obj/item/reagent_containers/cooking/proc/get_content_info()
+	var/string = "It contains:</br><ul><li>"
+	string += jointext(contents, "</li><li>") + "</li></ul>"
+	return string
+
+/obj/item/reagent_containers/cooking/proc/get_usable_status()
+	if(claimed)
+		return PCWJ_CONTAINER_BUSY
+	if(tracker)
+		return PCWJ_CONTAINER_BUSY
+	if(length(contents) || reagents.total_volume > 0)
+		return PCWJ_CONTAINER_BUSY
+
+	return PCWJ_CONTAINER_AVAILABLE
+
+/obj/item/reagent_containers/cooking/item_interaction(mob/living/user, obj/item/used, list/modifiers)
+	if(!tracker && (contents.len || reagents.total_volume != 0))
+		to_chat(user, "The [src] is full. Empty its contents first.")
+		return ITEM_INTERACT_COMPLETE
+
+	process_item(user, used)
+
+	return ITEM_INTERACT_COMPLETE
+
+/obj/item/reagent_containers/cooking/proc/process_item(mob/user, obj/used)
```

Comment:
`obj/used` doesn't provide a ton of info as to what it is, especially without proc documentation. I'm not asking you to document all these procs, but can you change the variable name/typepath to imply a bit clearer what it's supposed to be?

Reply:
Yeah, this parameter is a little vague because of how wide-ranged it can be. Honestly improving the documentation is the best thing to do here.